src/Sentry/SpanTracer.cs

```diff
@@ -30,6 +31,16 @@ public class SpanTracer : ISpan
     /// <inheritdoc />
     public bool IsFinished => EndTimestamp is not null;
 
+    // Not readonly because of deserialization
+    private readonly Lazy<Dictionary<string, Measurement>> _measurements = new();
```

Comment:
when are we using `Lazy` vs `==?` ? what's the criteria?

Reply:
In SpanTracer I used it because it was convenient... measurements only ever gets initialised from one place and it doesn't get assigned any items when it's being initialized (as above). 

In `Span` and `Transaction` it was a bit trickier because there are two different constructors. One of them doesn't include measurements and so could potentially be initialised as above. The other has measurements passed in:
https://github.com/getsentry/sentry-dotnet/blob/feat/4.0.0-remove-ihasmeasurements/src/Sentry/Span.cs#L103

It was messy to try to use `Lazy<T>` here - it seemed easier to use `??=`. I guess it could be done by passing a function to the Lazy<T> constructor instead:

```
_measurements = new Lazy<Dictionary<string, Measurement>>(() => tracer.Measurements.ToDictionary());
```

Maybe that's cleaner in hindsight.

Reply:
OK, I've just gone with Lazy<T> everywhere. I think I was just too lazy to do this in the first place 😜